Shopify/cli
# [Feature] Flow extensions payload preview

### WHY are these changes introduced?

This is a feature we're porting over from the Partner's Dashboard. The payload preview helps partners understand the shape of the data they'll either have to send to (triggers) or they'll receive (actions) from Flow.

In the case of triggers it is particularly crucial that we provide that information to partners since they need the `trigger_id` in order to make their request to Flow.

### WHAT is this pull request doing?

Adding a `Flow` service with a `preview` script which builds a preview JSON object for actions and a GraphQL mutation object for triggers.

That logic is then used during deployment and files are written to the Flow extensions that are being deployed.

Ideally I think we should also add that logic to the `app dev` command as well so developers can get the preview before deploying. Will continue exploring that.

### How to test your changes?

Generate a Flow extension, update your TOML config file and then trigger a deploy. A preview file should be added to your extension's directory and it should contain the correct information regarding your extension (e.g. config fields data, etc).

### Checklist
